dimod/binary_quadratic_model.py

```diff
@@ -526,64 +543,34 @@ def add_interaction(self, u, v, bias, vartype=None):
         quadratic = self.quadratic
         adj = self.adj
 
+        # so that they exist
+        self.add_variable(u, 0.0)
+        self.add_variable(v, 0.0)
+
         if vartype is not None and vartype is not self.vartype:
             if self.vartype is Vartype.SPIN and vartype is Vartype.BINARY:
                 # convert from binary to spin
-                bias /= 4.
+                bias /= 4
```

Comment:
Thanks for removing the hanging dots here and below. My eyes are thankful. :)

Reply:
If the biases are floats it actually speeds things up to have the dots! But it also would promote types to floats which is bad behaviour.